frontend/js/src/settings/brainzplayer/BrainzPlayerSettings.tsx

```diff
@@ -0,0 +1,216 @@
+          message={
+            <>
+              {error.toString()}
+              <br />
+              Please try again or contact us if the issue persists.
+            </>
+          }
+        />
+      );
+    }
+  }, [
+    youtubeEnabled,
+    spotifyEnabled,
+    soundcloudEnabled,
+    APIService,
+    currentUser?.auth_token,
+  ]);
+
+  return (
+    <>
+      <Helmet>
+        <title>BrainzPlayer Settings</title>
+      </Helmet>
+      <h2 className="page-title">BrainzPlayer Settings</h2>
+      <h4 className="page-title">Music Players</h4>
+      <p>
+        You can choose which music services to play music from on listenBrainz.
+        To sign in and authorize your account please go to the{" "}
+        <Link to="/settings/music-services/">music services page</Link>. <br />
+        If you don&apos;t have a subscription to a music service listed below,
+        Youtube will be used by default as it does not require an account.
+        However the search results from Youtube are often inaccurate, and for a
+        better listening experience we recommend one of the other services (if
+        you have one available).
+      </p>
+      <p>
+        <Switch
+          id="enable-spotify"
+          disabled={!SpotifyPlayer.hasPermissions(spotifyAuth)}
+          value="spotify"
+          checked={spotifyEnabled}
+          onChange={(e) => setSpotifyEnabled(!spotifyEnabled)}
+          switchLabel={
+            <span
+              className={`text-brand ${spotifyEnabled ? "text-success" : ""}`}
+            >
+              <FontAwesomeIcon icon={faSpotify} />
+              &nbsp;Spotify
+            </span>
+          }
+        />
+        <br />
+        <small>
+          Spotify requires a premium account to play full songs on third-party
+          websites. To sign in, please go to the{" "}
+          <Link to="/settings/music-services/">music services page</Link>.
+        </small>
+      </p>
+      {/* <p>
+        <Switch
+          id="enable-apple-music"
+          value="apple-music"
+          disabled={!AppleMusicPlayer.hasPermissions(appleAuth)}
+          checked={appleMusicEnabled}
+          onChange={(e) => setAppleMusicEnabled(!appleMusicEnabled)}
+          switchLabel={
+            <span className={`text-brand ${enabled ? "text-success":""}`}>
+              <FontAwesomeIcon icon={faApple} />
+              &nbsp;Apple Music Player
+            </span>
+          }
+        />
+        <br/>
+        <small>
+          Apple Music requires a premium account to play full songs on
+          third-party websites. To sign in, please go to the{" "}
+          <Link to="/settings/music-services/">music services page</Link>. You
+          will need to sign in every 6 months.
+        </small>
+      </p> */}
+      <p>
+        <Switch
+          id="enable-soundcloud"
+          value="soundcloud"
+          disabled={!SoundcloudPlayer.hasPermissions(soundcloudAuth)}
```

Comment:
Can we have a tooltip or something similar to inform the users that the switch has been disabled because they haven't authenticated their account?